src/Control/Monad/Logic/Sequence/Internal.hs

```diff
@@ -305,6 +319,47 @@ observeAll :: Seq a -> [a]
 observeAll = runIdentity . observeAllT
 {-# INLINE observeAll #-}
 
+-- | Convert @'SeqT' m a@ to @t m a@ when @t@ is some other logic monad
+-- transformer.
+toMonadLogicT :: (Monad m, Monad (t m), MonadTrans t, Alternative (t m)) => SeqT m a -> t m a
+toMonadLogicT (toView -> m) = lift m >>= \r -> case r of
+  Empty -> empty
+  a :< s -> pure a <|> toMonadLogicT s
```

Comment:
I'm not a huge fan of the name of this function. Any better ideas? Also, it suggests another function of type

```haskell
otherFunction :: (Monad m, Monad n, Alternative n) => (forall x. m a -> n a) -> SeqT m a -> n a
otherFunction phi (toView -> m) = phi m >>= \r -> case r of
  Empty -> empty
  a :< s -> pure a <|> otherFunction phi s
```
so that
```haskell
toMonadLogicT = otherFunction lift
```

Do we want that too, or is it too specialized?

Reply:
I'd probably call the first one `fromSeqT`.  What is the purpose of `forall x`? Did you mean to write `(forall x. m x -> n x)`?

Maybe `otherFunction` could be called `generalize`? I don't have a lot of good ideas on the name for that one. I think `streamly` has something similar but slightly more constrained like that in our types would be like `SeqT m a -> SeqT n a`, but I can't remember what they call it.

Ah found it: https://hackage.haskell.org/package/streamly-0.8.0/docs/Streamly-Internal-Data-Fold.html#v:hoist
```haskell
hoist :: (forall x. m x -> n x) -> Fold m a b -> Fold n a b
```
Your type is more general, but it's a pretty similar concept.

Reply:
Yes, I meant `forall x. m x -> n x`. `hoist` is something we need to consider too, but it's different!